tree/ntuple/v7/src/RPageStorage.cxx

```diff
@@ -264,14 +263,22 @@ ROOT::Experimental::Detail::RPageSink::SealPage(
    auto zippedBytes = packedBytes;
 
    if ((compressionSetting != 0) || !element.IsMappable()) {
-      zippedBytes = fCompressor->Zip(buffer, packedBytes, fOptions.GetCompression());
+      zippedBytes = compressor.Zip(buffer, packedBytes, compressionSetting);
```

Comment:
Oh, here we anyway had a lingering bug not picking the passed `compressionSetting`. Nice catch.